torchvision/datasets/standford_cars.py

```diff
@@ -0,0 +1,145 @@
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ) -> None:
+
+        try:
+            from scipy.io import loadmat
+
+            self._loadmat = loadmat
+        except ImportError:
+            raise RuntimeError("Scipy is not found. This dataset needs to have scipy installed: pip install scipy")
+
+        super().__init__(root, transform=transform, target_transform=target_transform)
+
+        self.train = train
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._samples = self._make_dataset()
+        self.classes = self._get_classes_name()  # class_id to class_name mapping
+
+    def _get_class_names(self) -> dict:
+        """
+        Returns Mapping of class ids to class names in form of Dictionary
+        """
+        meta_data = self._loadmat(os.path.join(self.root, "devkit/cars_meta.mat"))
+        class_names = meta_data["class_names"][0]
+        return {class_name[0].replace(" ", "_").replace("/", "_"): i for i, class_name in enumerate(class_names)}
+
+    def _make_dataset(self):
+        """
+        Returns Annotations for training data and testing data
+        """
+        annotations = None
+        if self.train:
+            annotations = self._loadmat(os.path.join(self.root, "devkit/cars_train_annos.mat"))
+        else:
+            annotations = self._loadmat(os.path.join(self.root, "cars_test_annos_withlabels.mat"))
+        samples = []
+        annotations = annotations["annotations"][0]
+        for index in range(len(annotations)):
+            target = annotations[index][4][0, 0]
+            image_file = annotations[index][5][0]
+            # Beware: Stanford cars targets starts at 1
+            target = target - 1
+            samples.append((image_file, target))
+        return samples
+
+    def __len__(self) -> int:
+        return len(self._samples)
+
+    def __getitem__(self, idx: int) -> (Image, int):
+        """Returns pil_image and class_id for given index"""
+        image_file, target = self._samples[idx]
+        image_path = os.path.join(self.root, f"cars_{'train' if self.train else 'test'}", image_file)
+        pil_image = Image.open(image_path).convert("RGB")
+
+        if self.transform is not None:
+            pil_image = self.transform(pil_image)
+        if self.target_transform is not None:
+            target = self.target_transform(target)
+        return pil_image, target
+
+    def download(self) -> None:
+        if self._check_exists():
+            return
+        else:
+            download_and_extract_archive(
+                url=self.urls[self.train], download_root=self.root, extract_root=self.root, md5=self.md5s[self.train]
```

Comment:
`extract_root`  defaults to `download_root` if nothing else is passed. Thus, since we want both to be `self.root`, we don't need to pass anything for `extract_root`

```suggestion
                url=self.urls[self.train], download_root=self.root, md5=self.md5s[self.train]
```

Please also apply this suggestion to the other downloads.